diff -Nurb man-1.6e.old/configure man-1.6e/configure
--- man-1.6e.old/configure	2006-05-01 18:56:14.000000000 +0100
+++ man-1.6e/configure	2007-02-20 23:02:31.000000000 +0000
@@ -476,18 +476,18 @@
   then
     if test $Fnroff = "missing"
     then
-      nroff="nroff -Tlatin1 -mandoc"
+      nroff="nroff -c -mandoc"
     else
-      nroff="$Fnroff -Tlatin1 -mandoc"
+      nroff="$Fnroff -c -mandoc"
     fi
     troff="troff -mandoc"
     echo "Warning: could not find groff"
   else
     if test $Fnroff = "missing"
     then
-      nroff="$Fgroff -Tlatin1 -mandoc"
+      nroff="$Fgroff -c -mandoc"
     else
-      nroff="$Fnroff -Tlatin1 -mandoc"
+      nroff="$Fnroff -c -mandoc"
     fi
     troff="$Fgroff -Tps -mandoc"
     jnroff="$Fgroff -Tnippon -mandocj"
