--- traceroute-1.4a12.old/Makefile.in	2004-08-16 13:23:51.000000000 +0100
+++ traceroute-1.4a12/Makefile.in	2004-08-16 13:23:51.000000000 +0100
@@ -90,7 +90,7 @@
 	sed -e 's/.*/char version[] = "&";/' $(srcdir)/VERSION > $@
 
 install: force
-	$(INSTALL) -m 4755 traceroute $(DESTDIR)/bin
+	$(INSTALL) -D -m 4755 traceroute $(DESTDIR)/bin/traceroute
 
 install-man: force
 	$(INSTALL) traceroute.8 $(DESTDIR)$(mandir)/man8
