diff -Nurb linux-2.6.16.52.old/drivers/scsi/sata_via.c linux-2.6.16.52/drivers/scsi/sata_via.c
--- linux-2.6.16.52.old/drivers/scsi/sata_via.c	2007-05-30 21:36:59.000000000 +0100
+++ linux-2.6.16.52/drivers/scsi/sata_via.c	2007-07-24 16:01:19.000000000 +0100
@@ -76,6 +76,7 @@
 static void svia_scr_write (struct ata_port *ap, unsigned int sc_reg, u32 val);
 
 static const struct pci_device_id svia_pci_tbl[] = {
+	{ 0x1106, 0x0591, PCI_ANY_ID, PCI_ANY_ID, 0, 0, vt6420 },
 	{ 0x1106, 0x3149, PCI_ANY_ID, PCI_ANY_ID, 0, 0, vt6420 },
 	{ 0x1106, 0x3249, PCI_ANY_ID, PCI_ANY_ID, 0, 0, vt6421 },
 
