--- openntpd-3.9p1/a.c	2010-02-12 00:15:18.000000000 -0500
+++ openntpd-3.9p1/ntpd.c	2010-02-12 00:20:48.000000000 -0500
@@ -140,9 +140,16 @@
 
 	if (!conf.settime) {
 		log_init(conf.debug);
-		if (!conf.debug)
+		if (!conf.debug) {
 			if (daemon(1, 0))
 				fatal("daemon");
+			FILE *pidfile = fopen("/var/run/ntpd.pid", "w");
+			if (pidfile) {
+				fprintf(pidfile, "%d\n", getpid());
+				fclose(pidfile);
+			}
+		}
+
 	} else
 		timeout = SETTIME_TIMEOUT * 1000;
 
@@ -215,6 +222,9 @@
 	if (chld_pid)
 		kill(chld_pid, SIGTERM);
 
+	if (!conf.debug)
+		unlink("/var/run/ntpd.pid");
+
 	do {
 		if ((pid = wait(NULL)) == -1 &&
 		    errno != EINTR && errno != ECHILD)
