--- imspector-0.9/sslstate.cpp-orig	2011-12-26 23:23:36.000000000 -0500
+++ imspector-0.9/sslstate.cpp	2011-12-26 23:26:11.000000000 -0500
@@ -28,7 +28,7 @@
 	SSL_library_init();
 	SSL_load_error_strings();
 	
-	method = SSLv23_method();
+	method = (SSL_METHOD *) SSLv23_method();
 	
 	if (!method)
 	{
@@ -36,8 +36,8 @@
 		return false;
 	}
 	
-	connectctx = SSL_CTX_new(method);
-	ctx = SSL_CTX_new(method);
+	connectctx = SSL_CTX_new((const SSL_METHOD *) method);
+	ctx = SSL_CTX_new((const SSL_METHOD *) method);
 
 	if (!connectctx || !ctx)
 	{
