--- autogen.sh.old	2012-08-04 11:51:32.000000000 -0700
+++ autogen.sh	2012-08-05 09:44:09.000000000 -0700
@@ -435,15 +435,22 @@
 rm -f confdefs.h
 
 # Get nDPI
-
+set -x
 echo "9. Downloading nDPI..."
 
 NDPI_URL=https://svn.ntop.org/svn/ntop/trunk/nDPI/
+#NDPI_URL=https://131.114.21.22/svn/ntop/trunk/nDPI
 if test -d nDPI; then
     echo "nDPI already available"
 else
-    svn co $NDPI_URL
+    if test -e nDPI; then
+        rm -rf nDPI
+        mkdir nDPI
+    fi
+    curl --insecure $NDPI_URL
 fi
+set +x
+sleep 10
 
 dnl> nDPI compilation
 if test -f NDPI_LIB; then
