--- miniupnpd-20070605.new/genconfig.sh	2007-03-18 10:31:48.000000000 +0000
+++ miniupnpd-20070605/genconfig.sh	2007-06-12 15:54:39.000000000 +0100
@@ -16,8 +16,8 @@
 LOG_MINIUPNPD="LOG_DAEMON"
 
 # detecting the OS name and version
-OS_NAME=`uname -s`
-OS_VERSION=`uname -r`
+OS_NAME="SmoothWall Express"
+OS_VERSION=3.0
 
 # Debian GNU/Linux special case
 if [ -f /etc/debian_version ]; then
@@ -72,6 +72,9 @@
 	Linux)
 		OS_URL=http://www.kernel.org/
 		;;
+	"SmoothWall Express")
+		OS_URL=http://www.smoothwall.org/
+		;;
 	*)
 		echo "Unknown OS : $OS_NAME"
 		echo "Please contact the author at http://miniupnp.free.fr/"
