--- miniupnpd-1.7/Makefile.linux-ORIG	2012-12-16 01:54:50.000000000 -0500
+++ miniupnpd-1.7/Makefile.linux	2012-12-16 01:56:02.000000000 -0500
@@ -65,7 +65,7 @@
 ARCH ?= $(shell uname -m | grep -q "x86_64" && echo 64)
 ifdef IPTABLESPATH
 CFLAGS := $(CFLAGS) -I$(IPTABLESPATH)/include/
-LDFLAGS := $(LDFLAFGS) -L$(IPTABLESPATH)/libiptc/
+LDFLAGS := $(LDFLAFGS) -L$(IPTABLESPATH)/libip4tc/ -L$(IPTABLESPATH)/libip6tc/
 # get iptables version and set IPTABLES_143 macro if needed
 ifeq ($(TARGET_OPENWRT),)
 IPTABLESVERSION := $(shell grep "\#define VERSION" $(IPTABLESPATH)/config.h | tr -d \" |cut -d" " -f3 )
