--- squid-2.5.STABLE7/errors/English.old/ERR_CACHE_ACCESS_DENIED	2002-08-28 21:07:58.000000000 +0000
+++ squid-2.5.STABLE7/errors/English/ERR_CACHE_ACCESS_DENIED	2005-01-06 15:57:36.000000000 +0000
@@ -1,34 +1,3 @@
-<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
-<HTML><HEAD><META HTTP-EQUIV="Content-Type" CONTENT="text/html; charset=iso-8859-1">
-<TITLE>ERROR: Cache Access Denied</TITLE>
-<STYLE type="text/css"><!--BODY{background-color:#ffffff;font-family:verdana,sans-serif}PRE{font-family:sans-serif}--></STYLE>
-</HEAD>
-<BODY>
-<H1>ERROR</H1>
-<H2>Cache Access Denied</H2>
-<HR noshade size="1px">
-<P>
-While trying to retrieve the URL:
-<A HREF="%U">%U</A>
-<P>
-The following error was encountered:
-<UL>
-<LI>
-<STRONG>
-Cache Access Denied.
-</STRONG>
-</UL>
-</P>
-
-<P>Sorry, you are not currently allowed to request:
-<PRE>    %U</PRE>
-from this cache until you have authenticated yourself.
-</P>
-
-<P>
-You need to use Netscape version 2.0 or greater, or Microsoft Internet
-Explorer 3.0, or an HTTP/1.1 compliant browser for this to work.  Please
-contact the <A HREF="mailto:%w">cache administrator</a> if you have
-difficulties authenticating yourself or 
-<A HREF="http://%h/cgi-bin/chpasswd.cgi">change</a> your default password.
-</P>
+<HTML><HEAD><TITLE>ERROR: Cache Access Denied</TITLE></HEAD><BODY><CENTER><TABLE WIDTH=710 BGCOLOR=#dddddd><TR><TD>&nbsp;</TD></TR><TR><TD><TABLE WIDTH=99% ALIGN=CENTER CELLPADDING=0 BGCOLOR=#cccccc CELLSPACING=0><TR HEIGHT=1><TD BGCOLOR=#ffffff></TD></TR><TR><TD ALIGN=CENTER><h2>ERROR: Cache Access Denied</h2>Sorry, you are not currently allowed to request:<PRE><A HREF=%U>%U</A>
+</PRE>from this cache because your username and password are incorrect or your user group does not have permission to web browse. Please check your username and password. They are case sensitive so make sure Caps Lock is not on.
+</TD></TR><TR HEIGHT=1><TD BGCOLOR=#888888></TD></TR></TABLE></TD></TR><TR><TD>&nbsp;</TD></TR></TABLE></CENTER>
